### PR TITLE
chore: Improve dropdown draggable list

### LINKED ISSUE #1668: Improve Dropdown Draggable list
## Scope and Context

In previous PRs: https://github.com/twentyhq/twenty/pull/1667 and https://github.com/twentyhq/twenty/pull/1636, we have introduced a new MenuItem draggable in the dropdown and implemented a drag and drop behavior.

This is working but it would be great to refactor this into separated components so we can re-use them

### MenuItemDraggable

Right now we are using `MenuItem` with `isDraggable` prop to create the DraggableItem
Instead, we want to create a separated component `MenuItemDraggable`, remove `isDraggable` prop from `MenuItem` and `MenuItemLeftComponent`.
We will also need to add a `isDragged` prop to this `MenuItemDraggable` to force the "hover" look while the item is being dragged.

We also want to add storybook tests on `MenuItemDraggable` (like we have on `MenuItemNavigate` for example)

### DraggableList

Having a list that is draggable will be useful, not only in dropdown.
- Create a folder `@/ui/draggable-list `with a `DraggableList` component
- This component should take as prop: `itemsComponents`, `onDragEnd((previousIndex, nextIndex) => {})`
- Use this component in `ViewFieldsVisibilityDropdownSection` (move the logic from ViewFieldsVisibilityDropdownSection to `DraggableList`) by passing a list of `DraggableMenuItems`

Add a storybook test on this list (we don't know how to actually test the draggable behavior, but we can at least make sure the component renders correctly a list of items)

### PR DESCRIPTION
**Pull request by AdityaPimpalkar**
closes: #1668

Feedback is appreciated! :)

### CODE REVIEW

File: front/src/modules/ui/menu-item/components/MenuItemDraggable.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+}: MenuItemDraggableProps) => {
+  const showIconButtons = Array.isArray(iconButtons) && iconButtons.length > 0;
+  const theme = useTheme();
+
+  return (
+    <StyledHoverableMenuItemBase
+      data-testid={key ?? undefined}
+      onClick={onClick}
+      accent={accent}
+      className={isDragged ? `hover` : className}
```
Comment (AdityaPimpalkar) [inline front/src/modules/ui/menu-item/components/MenuItemDraggable.tsx:46]: The hover class is not getting triggered, should I pass it as a parameter to `StyledHoverableMenuItemBase`?

**Review by AdityaPimpalkar — COMMENTED** [review]
CSS change on dragging.

**Review by charlesBochet — COMMENTED** [review]
Thank you, this will greatly improve the API of twenty by providing draggable lists

File: front/src/modules/ui/draggable-list/components/DraggableItem.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                top: 'auto',
+                transform: draggableStyle?.transform?.replace(
+                  /\(-?\d+px,/,
+                  '(0,',
+                ),
+              },
+            }}
+            {...draggableProvided.dragHandleProps}
+          >
+            {React.cloneElement(itemsComponent, { isDragged })}
```
[the remark this answers is not in the document]
Comment (AdityaPimpalkar) [reply front/src/modules/ui/draggable-list/components/DraggableItem.tsx]: Shouldn't be needing this now (see comment below) :)

File: front/src/modules/ui/menu-item/components/MenuItemDraggable.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    >
+      <StyledMenuItemLeftContent>
```
[the remark this answers is not in the document]
Comment (AdityaPimpalkar) [reply front/src/modules/ui/menu-item/components/MenuItemDraggable.tsx]: You're right!

File: front/src/modules/ui/draggable-list/components/DraggableItem.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                ...draggableStyle,
+                left: 'auto',
+                top: 'auto',
+                transform: draggableStyle?.transform?.replace(
+                  /\(-?\d+px,/,
+                  '(0,',
+                ),
+                background: isDragged
+                  ? theme.background.transparent.light
+                  : 'none',
```
Comment (AdityaPimpalkar) [inline front/src/modules/ui/draggable-list/components/DraggableItem.tsx:46]: I found the css fix on `DraggableItem`

**Review by lucasbordeau — APPROVED** [review]

**Comment by charlesBochet** [comment]
@AdityaPimpalkar Thanks a lot, I have made a few small changes after discussing with the team here: https://github.com/twentyhq/twenty/pull/1874

**Pull request merged**